CMakeLists.txt

```diff
@@ -25,19 +98,19 @@ find_package(OpenMP)
 ###########
 ## Build ##
 ###########
-
 ament_auto_add_library(ndt_omp SHARED
   src/pclomp/voxel_grid_covariance_omp.cpp
   src/pclomp/ndt_omp.cpp
   src/pclomp/gicp_omp.cpp
 )
 
-target_link_libraries(ndt_omp PUBLIC ${PCL_LIBRARIES})
+target_link_libraries(ndt_omp ${PCL_LIBRARIES})
 
 if(OpenMP_CXX_FOUND)
-  target_link_libraries(ndt_omp PUBLIC OpenMP::OpenMP_CXX)
+  target_link_libraries(ndt_omp OpenMP::OpenMP_CXX)
```

Comment:
Why did you remove `PUBLIC`?

Reply:
This is because the docker CI for ROS2 failed with the following error.
And the solution to this error is to remove `PUBLIC`.
<https://stackoverflow.com/questions/59522267/cmake-rejects-a-second-target-link-libraries-talking-about-keyword-vs-plain>

```
Starting >>> ndt_omp
--- stderr: ndt_omp
CMake Warning (dev) at CMakeLists.txt:87 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable PCL_ROOT is set to:

    /usr

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at CMakeLists.txt:107 (target_link_libraries):
  The plain signature for target_link_libraries has already been used with
  the target "ndt_omp".  All uses of target_link_libraries with a target must
  be either all-keyword or all-plain.

  The uses of the plain signature are here:

   * /opt/ros/foxy/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:145 (target_link_libraries)



---
Failed   <<< ndt_omp [9.05s, exited with code 1]
```

Reply:
Is this because of `ament_auto_add_library`?
I feel `all-keyword` is better if possible.

Reply:
Will this work?
https://github.com/ament/ament_cmake/blob/4ac1ed1153ac774116c273021944eb3ea836495c/ament_cmake_auto/cmake/ament_auto_add_library.cmake#L66

```cmake
ament_auto_add_library(ndt_omp SHARED PUBLIC
  src/pclomp/voxel_grid_covariance_omp.cpp
  src/pclomp/ndt_omp.cpp
  src/pclomp/gicp_omp.cpp
)

target_link_libraries(ndt_omp PUBLIC ${PCL_LIBRARIES})

if(OpenMP_CXX_FOUND)
  target_link_libraries(ndt_omp PUBLIC OpenMP::OpenMP_CXX)
```

Reply:
Ah, but since the upstream's style is `plain`, I think it's okay.
https://github.com/koide3/ndt_omp/blob/399d61318a471758234e0d747ab4e407c74fcb76/CMakeLists.txt

Reply:
> Will this work? https://github.com/ament/ament_cmake/blob/4ac1ed1153ac774116c273021944eb3ea836495c/ament_cmake_auto/cmake/ament_auto_add_library.cmake#L66
> 
> ```cmake
> ament_auto_add_library(ndt_omp SHARED PUBLIC
>   src/pclomp/voxel_grid_covariance_omp.cpp
>   src/pclomp/ndt_omp.cpp
>   src/pclomp/gicp_omp.cpp
> )
> 
> target_link_libraries(ndt_omp PUBLIC ${PCL_LIBRARIES})
> 
> if(OpenMP_CXX_FOUND)
>   target_link_libraries(ndt_omp PUBLIC OpenMP::OpenMP_CXX)
> ```

It doesn't work. :cry: 

> Ah, but since the upstream's style is plain, I think it's okay.

Thanks for letting me know. I'll keep the current code.

Reply:
:pleading_face: 
Okay, thank you!